components/feature/tabs/src/main/java/org/mozilla/rocket/tabs/TabViewEngineObserver.kt

```diff
@@ -83,7 +83,7 @@ class TabViewEngineObserver(
 
     override fun onExternalResource(
         url: String,
-        fileName: String?,
+        fileName: String,
```

Comment:
it seems the caller of this method get all parameters from a Download object, which declares its name field as nullable, is it safe here to change it to non-optional type?

Reply:
After upgrading android-components, they changed its `fileName` field to non-null from `EngineSession.Observer`